src/Features/Core/Portable/EditAndContinue/CommittedSolution.cs

```diff
@@ -124,6 +125,9 @@ public bool HasNoChanges(Solution solution)
     public Project GetRequiredProject(ProjectId id)
         => _solution.GetRequiredProject(id);
 
+    public bool IsStaleProject(ProjectId id)
+        => _staleProjects.Contains(id);
```

Comment:
_staleProjects is mutated under a lock below.  but there is no lock here.  this feels unsafe.  if it's not, please include docs in the code why this is ok.

Reply:
Ah, yeah, should be locked.